packages/datadog_gql_link/lib/src/datadog_gql_link.dart

```diff
@@ -0,0 +1,234 @@
+    listener?.requestStarted(request, userAttributes);
+    request = _injectTracingHeaders(request);
+
+    final resourceId = _startRumResource(
+        request, internalAttributes, tracingContext, userAttributes);
+
+    return forward!(request).transform(StreamTransformer.fromHandlers(
+      handleData: (data, sink) {
+        listener?.responseRecieved(data, userAttributes);
+
+        var linkResponseContext = data.context.entry<HttpLinkResponseContext>();
+        int? statusCode;
+        int? size;
+        if (linkResponseContext != null) {
+          statusCode = linkResponseContext.statusCode;
+          final contentLength = linkResponseContext.headers?['content-length'];
+          if (contentLength != null) {
+            size = int.tryParse(contentLength);
+          }
+        }
+
+        final errorMap = _serializeResponseErrors(data);
+
+        datadogSdk.rum?.stopResourceLoading(
+          resourceId,
+          statusCode,
+          RumResourceType.native,
+          size,
+          {
+            if (errorMap != null) ...errorMap,
+            ...userAttributes,
+          },
+        );
+
+        sink.add(data);
+      },
+      handleError: (error, stackTrace, sink) {
+        listener?.requestError(error, stackTrace, userAttributes);
+        datadogSdk.rum?.stopResourceLoadingWithErrorInfo(resourceId,
+            error.toString(), error.runtimeType.toString(), userAttributes);
+
+        sink.addError(error, stackTrace);
+      },
+    ));
+  }
+
+  Map<String, String> _getInternalAttributes(Request request) {
+    final attributes = <String, String>{};
+
+    final operationType = request.operation.getOperationType();
+    if (operationType != null) {
+      switch (operationType) {
+        case OperationType.mutation:
+          attributes[_GraphQLAttributes.operationType] = 'mutation';
+          break;
+        case OperationType.subscription:
+          attributes[_GraphQLAttributes.operationType] = 'subscription';
+          break;
+        case OperationType.query:
+          attributes[_GraphQLAttributes.operationType] = 'query';
+          break;
+      }
+    }
+
+    final operationName = request.operation.operationName;
+    if (operationName != null) {
+      attributes[_GraphQLAttributes.operationName] = operationName;
+    }
+
+    return attributes;
+  }
+
+  String _startRumResource(
+      Request request,
+      Map<String, String> internalAttributes,
+      TracingContext? tracingContext,
+      Map<String, Object?> userAttributes) {
+    final resourceId = _uuid.v1();
+    final datadogAttributes = generateDatadogAttributes(
+        tracingContext, datadogSdk.rum?.tracingSamplingRate ?? 0);
+    final attributes = {
+      ...userAttributes,
+      ...datadogAttributes,
+      '_dd': {
+        'graphql': {
+          ...internalAttributes,
+        }
+      },
+    };
+
+    // TODO: Assume `post` for now, but most links support `get` queries.
```

Comment:
it is better to add a ticket reference here, so that we don't forget

Reply:
Yes, good call. I'll add it.